### DISCUSSION TITLE
Python version conflicts - extended installation

### DISCUSSION THREAD

**Discussion by nbeine** [SESMG/SESMG#204] [Ideas]
Hello everyone,

As the extended installation is recommended in the installation guide, I carried that one out.
Even though the working python versions (3.7 - 3.9) are stated in guide, there may emerge conflicts with the installation:

I had already installed python on my computer (as version 3.10). This version (and newer versions as well) trigger conflicts with the h5py version 3.6.0, which is demanded in the requirements.txt. 

My idea/suggestion is the following:

Even though you state the possible python versions in your manual, it may be advantageous to change the way you query the version in the developer installation. When entering 3.10 (or newer) in the developer installation command, the installation may be aborted and the user may receive a warning, that the installed python version leads to conflicts. This may save a lot of time when searching the error cause.

Like stated before, you mention the working versions. But maybe stress the possible conflicts more or build in a query/warning. 

Cheers,
@nbeine

**Comment by jtock** [comment]
Hi @nbeine, 
thank you for your helpful idea and contribution. Sorry, that you spent that much time to find a solution. 
I opened an issue based on your idea that we have it on our to do list. We will implement anything as an additional step to prevent users from installing the SESMG with an unsupported Python version.
Feel free to use the discussion section again if you have any questions while using the SESMG!
Thanks again and greetings 
Jan
